src/view.c

```diff
@@ -658,6 +659,21 @@ void
 view_set_decorations(struct view *view, bool decorations)
 {
 	assert(view);
+
+	/* Window-rules take priority if they exist for this view */
+	switch (window_rules_get_property(view, "serverDecoration")) {
+	case LAB_PROP_TRUE:
+		view->ssd_enabled = true;
+		decorate(view);
+		return;
+	case LAB_PROP_FALSE:
+		view->ssd_enabled = false;
+		undecorate(view);
+		return;
+	default:
+		break;
+	}
+
```

Comment:
I am not sure if we should use `view_set_decorations` to implement this. I didn't test it but I'd assume this will stop `ToggleDecorations` to work. If that is the case we should likely just set the initial `view->ssd_enabled` bool on map.
Although I am not sure right now how that interacts with the 2 decoration protocols.

Reply:
It depends on if we want the property to *always* apply - or only apply when the view is first mapped.

I thought that the always (persistent) approach was best - but don't mind.

Reply:
I'd argue that we shouldn't break manual `ToggleDecorations` actions by the user, even if some rule had been set.

Reply:
On second thought, agree. Have moved that code to has_ssd() and want_deco(). Seems to work fine.